api/collection.go

```diff
@@ -103,6 +111,60 @@ func (api *API) AddCollectionHandler(w http.ResponseWriter, r *http.Request) {
 	log.Event(ctx, "add collection request completed successfully", log.INFO, logData)
 }
 
+func (api *API) PutCollectionHandler(w http.ResponseWriter, req *http.Request) {
+	defer dphttp.DrainBody(req)
+
+	ctx := req.Context()
+	logData := log.Data{}
+	collectionID := mux.Vars(req)["collection_id"]
+	logData["collection_id"] = collectionID
+
+	log.Event(ctx, "put collection", log.INFO, logData)
+
+	err := ValidateUUID(collectionID)
+	if err != nil {
+		handleError(ctx, collections.ErrInvalidID, w, logData)
+		return
+	}
+
+	_, err = api.collectionStore.GetCollectionByID(ctx, collectionID, models.AnyETag)
+	if err != nil {
+		handleError(ctx, err, w, logData)
+		return
+	}
+
+	// eTag value must be present in If-Match header
+	eTag, err := getIfMatchForce(req)
+	if err != nil {
+		log.Event(ctx, "missing header", log.ERROR, log.Data{"error": err.Error()})
+		handleError(ctx, err, w, logData)
+		return
+	}
+
+	collection, err := ParseCollection(ctx, req.Body)
+	if err != nil {
+		handleError(ctx, err, w, logData)
+		return
+	}
+
+	collection.ID = collectionID
+
+	if err := api.collectionStore.ReplaceCollection(ctx, collection, eTag); err != nil {
+		handleError(ctx, err, w, logData)
+		return
+	}
+
+	setETag(w, collection.ETag)
+	w.WriteHeader(http.StatusOK)
+	err = WriteJSONBody(ctx, collection, w, logData)
+	if err != nil {
+		handleError(ctx, err, w, logData)
+		return
+	}
+
+	log.Event(ctx, "put collection request completed successfully", log.INFO, logData)
```

Comment:
Is there value in including the etag in the log data? Allows you to line up events/updates with the a specific etag value - might be useful if debugging issues.